units/db_cleanup_general.go

```diff
@@ -0,0 +1,101 @@
+package units
+
+import (
+	"context"
+	"time"
+
+	"github.com/evergreen-ci/evergreen"
+	"github.com/mongodb/grip/message"
+	"github.com/pkg/errors"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+const (
+	cleanupBatchSize = 100 * 1000
+)
+
+type FilterFunc func(time.Time) map[string]interface{}
+
+type DataCleanupJobBase struct {
+	CollectionName string        `bson:"collection_name" json:"collection_name" yaml:"collection_name"`
+	TTL            time.Duration `bson:"ttl" json:"ttl" yaml:"ttl"`
+	Errors         []error       `bson:"errors" json:"errors" yaml:"errors"`
```

Comment:
Errors are actually non-marshallable - they're more like an interface (for anything that implements the `Error` method) so you would have to store these as their string representations if you want them to persist in the DB.